### ISSUE TITLE
Community EE

### ISSUE THREAD

**Issue by anweshadas** [ansible-community/community-topics#267]
### Summary

- Feedback on this PR https://github.com/ansible-community/images/pull/61
- Where do we want to push the artifacts to `ansible/packages/container/package/community-ee` or `ansible-community//packages/container/package/community-ee` namespace?
- What are the steps that needs that needs to be done before we make it available to the public?

**Comment by samccann** [comment]
See the end part of the [community WG log](https://meetbot.fedoraproject.org/ansible-community/2023-08-16/ansible_community_meeting.2023-08-16-18.00.log.html) for discussion of this EE.

**Comment by felixfontein** [comment]
(I'll look at the PR later or the next days. I'm just writing about the general idea here.)

I think the main reservations I have with the "community EE" is it's name. I think calling it "community EE" raises a lot of different expectations for different folks, which a single EE can never satisfy (and a single EE can also not satisfy a large subset of these expectations). So picking a different name for it would already help a lot :) A first idea I had was something with "minimal", but a truly minimal EE only has ansible-core and no collection. Maybe "ansible EE" would be better, since this EE contains a lot of ansible.* stuff?

Another thing is that for most of my playbooks, I need *some* modules from community.* collections. So this specific EE would not be useful for me for almost everything I do. I don't know how it looks like for others (as usual: YMMV). This EE is probably fine to try some things out, but - you can also do that with a more minimal one that only contains ansible-core itself. As folks mentioned in the meeting, it would be more helpful to have smaller specifically themed EEs. @acozine mentioned a network EE, but similar to, for example, a "database EE" the question is what kind of collections you want to put in there. Having an EE with all kind of network collections would be pretty large (thinking of the large fortinet collection that's in Ansible). Having specific smaller collections (for example, a RouterOS EE, or a Cisco EE, or a (insert vendor name here) EE) might be better - but then we end up with a large set of EEs to maintain. Not sure how helpful that is.

**Comment by felixfontein** [comment]
Maybe it would also be helpful to allow collection maintainers to create, publish, and maintain small EEs with their collection and a few others they think would be useful for users of their own collection. For example, I'm thinking of a RouterOS collection which contains community.routeros (and its dependencies ansible.netcommon and thus also ansible.utils), and potentially also community.crypto. Maintaining such an EE myself is something I probably won't do (sounds like too much work, and infrastructure required for it). Having a simple way to do it - simply put the list of collections in some YAML file somewhere - would be different. I guess there are more community collection maintainers with similar reservations for maintaining and publishing EEs for their collections.

**Comment by samccann** [comment]
Summarizing the 'feel' of the community WG discussion:
1. let's not call it community EE. There was feeling the name suggests it's either the entire Ansible package, or at least community.xx collections and the current iteration is all ansible.xx collections.
2. Could we group-design a 'better' first EE to offer the community? As in an EE that has the most popular collections in it (tho hard to find what those are).

**Comment by felixfontein** [comment]
Regarding naming for the one in the PR: how about "simple EE"? Simple doesn't imply minimal, and it doesn't imply that it contains a certain/larger set of useful collections.

**Comment by felixfontein** [comment]
> Where do we want to push the artifacts to `ansible/packages/container/package/community-ee` or `ansible-community//packages/container/package/community-ee` namespace?

Are these paths relative to the GitHub Container Registry (GHCR)? Or to quay.io? The community ansible-test images are hsoted on quay.io (though I have no idea who actually owns and manages the namespace where they live) and they are built by Zuul. I don't know which container registry is preferrable, or which one is easier to use/manage/...

**Comment by gotmax23** [comment]
See https://forum.ansible.com/t/community-execution-environment-image-content-and-location/262

**Comment by Andersson007** [comment]
Closing the topic. Let's continue on the [forum](https://forum.ansible.com/) in topics under exec-env tag.
Thanks everyone! (feel free to reopen this one if you think something is undone within it)
